habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -470,6 +470,218 @@ def step(self, *args, is_last_action, **kwargs):
+        super().__init__(*args, config=config, sim=sim, **kwargs)
+        self._sim: RearrangeSim = sim
+        self.base_vel_ctrl = habitat_sim.physics.VelocityControl()
+        self.base_vel_ctrl.controlling_lin_vel = True
+        self.base_vel_ctrl.lin_vel_is_local = True
+        self.base_vel_ctrl.controlling_ang_vel = True
+        self.base_vel_ctrl.ang_vel_is_local = True
+        self._allow_dyn_slide = self._config.get("allow_dyn_slide", True)
+        self._allow_back = self._config.allow_back
+        self._collision_threshold = self._config.collision_threshold
+        self._longitudinal_lin_speed = self._config.longitudinal_lin_speed
+        self._lateral_lin_speed = self._config.lateral_lin_speed
+        self._ang_speed = self._config.ang_speed
+        self._navmesh_offset = self._config.navmesh_offset
+        self._enable_lateral_move = self._config.enable_lateral_move
+
+    @property
+    def action_space(self):
+        lim = 20
+        if self._enable_lateral_move:
+            return spaces.Dict(
+                {
+                    self._action_arg_prefix
+                    + "base_vel": spaces.Box(
+                        shape=(3,), low=-lim, high=lim, dtype=np.float32
+                    )
+                }
+            )
+        else:
+            return spaces.Dict(
+                {
+                    self._action_arg_prefix
+                    + "base_vel": spaces.Box(
+                        shape=(2,), low=-lim, high=lim, dtype=np.float32
+                    )
+                }
+            )
+
+    def collision_check(
+        self, trans, target_trans, target_rigid_state, compute_sliding
+    ):
+        """
+        trans: the transformation of the current location of the robot
+        target_trans: the transformation of the target location of the robot given the center original Navmesh
+        target_rigid_state: the target state of the robot given the center original Navmesh
+        compute_sliding: if we want to compute sliding or not
+        """
+        num_check_cylinder = len(self._navmesh_offset)
+        # Get the offset positions
+        cur_pos = []
+        goal_pos = []
+        for i in range(num_check_cylinder):
+            cur_pos.append(
+                trans.transform_point(
+                    np.array(
+                        [
+                            self._navmesh_offset[i][0],
+                            0,
+                            self._navmesh_offset[i][1],
+                        ]
+                    )
+                )
+            )
+            goal_pos.append(
+                target_trans.transform_point(
+                    np.array(
+                        [
+                            self._navmesh_offset[i][0],
+                            0,
+                            self._navmesh_offset[i][1],
+                        ]
+                    )
+                )
+            )
```

Comment:
Nit: slightly more concise. You'll need to re-run pre-commit for format correction.
```suggestion
nav_pos_3d = [np.array([xz[0], 0, xz[1]]) for xz in self._navmesh_offset]
cur_pos = [trans.transform_point(xyz) for xyz in nav_pos_3d]
goal_pos = [target_trans.transform_point(xyz) for xyz in nav_pos_3d]
```